src/config.py

```diff
@@ -40,6 +40,12 @@ def get_logs_dir(self):
     def get_repos_dir(self):
         return self.config["STORAGE"]["REPOS_DIR"]
     
+    def get_openai_base_url(self):
+        return self.config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
+    
+    def get_search_engine(self):
+        return self.config["USER_PREFS"]["SEARCH_ENGINE"]
```

Comment:
**Suggestion**: Add error handling or validation for the "get_search_engine" and "get_openai_base_url" methods to ensure the configurations are valid and exist within the config file. This will prevent runtime errors due to missing configurations. <br> **Code Suggestion**: 
 ```
 +        try:
 +            return self.config["API_ENDPOINTS"]["OPENAI_BASE_URL"]
 +        except KeyError:
 +            # Log error or set a default value
 ```